File: templates/r/histogram.R
## 直方图（+核密度叠加）—— 卡片 01_分布图/直方图_Histogram.md
## 合成数据，可独立运行： Rscript templates/r/histogram.R
.args <- commandArgs(trailingOnly = FALSE)
.sd <- dirname(normalizePath(sub("^--file=", "", .args[grep("^--file=", .args)])))
source(file.path(.sd, "_common.R"))
set.seed(2026)

# >>> PARAM ------------------------------------------------------
NAME <- "tpl_histogram"; TITLE <- "Distribution of a continuous variable"
XLAB <- "Value"; YLAB <- "Density"
FILL <- pal("cat_main")[2]; LINE <- pal("cat_main")[5]   # 浅蓝填充 + 深蓝密度线
BINS <- 40
# 换真实数据： df <- readr::read_csv("data.csv"); 列 x
df <- data.frame(x = rnorm(2000, 230, 45))
# <<< PARAM ------------------------------------------------------

p <- ggplot(df, aes(x)) +
  geom_histogram(aes(y = after_stat(density)), bins = BINS, fill = FILL,
                 color = "white", alpha = .85) +
  geom_density(color = LINE, linewidth = 1) +
  labs(x = XLAB, y = YLAB, title = TITLE)
save_gg(p, NAME, w = 92, h = 72)

File: templates/python/histogram.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-
"""直方图（+核密度叠加）—— 卡片 01_分布图/直方图_Histogram.md
合成数据，可独立运行。改 # >>> PARAM 区换数据/分箱/配色。"""
import os, sys, warnings; warnings.filterwarnings("ignore")
sys.path.insert(0, os.path.dirname(os.path.abspath(__file__)))
import numpy as np, seaborn as sns, matplotlib.pyplot as plt
from _common import set_theme, save_fig, pal
set_theme(); np.random.seed(7)

# >>> PARAM ------------------------------------------------------
NAME = "tpl_histogram"
TITLE = "Distribution of a continuous variable"; XLAB = "Value"; YLAB = "Density"
FILL = pal("cat_main")[1]; LINE = pal("cat_main")[4]   # 浅蓝填充 + 深蓝密度线
BINS = 40
FIGSIZE = (3.6, 2.9)
# 换真实数据： x = df["TOTCHOL"].dropna().values
x = np.random.normal(230, 45, 2000)
# <<< PARAM ------------------------------------------------------

fig, ax = plt.subplots(figsize=FIGSIZE)
sns.histplot(x, bins=BINS, stat="density", color=FILL, edgecolor="white",
             alpha=.85, ax=ax)
sns.kdeplot(x, color=LINE, lw=1.6, ax=ax)
ax.set_title(TITLE); ax.set_xlabel(XLAB); ax.set_ylabel(YLAB)
save_fig(fig, NAME)
